core/src/main/java/io/grpc/MessageEncoding.java

```diff
@@ -39,12 +45,13 @@ public OutputStream compress(OutputStream os) throws IOException {
     }
   };
 
-  private static final ConcurrentMap<String, Decompressor> decompressors =
+  private static final ConcurrentMap<String, DecompressorInfo> decompressors =
       initializeDefaultDecompressors();
 
   /**
    * Represents a message compressor.
    */
+  @ExperimentalApi
```

Comment:
Not sure we need `@ExperimentalApi` for every inner class, given that the outer class is already marked as such.  WDYT?